src/http-adapter.js

```diff
@@ -112,36 +123,30 @@ class HttpAdapter {
     return promise
```

Comment:
This works because of the async/await unwrap logic, but I think we should do an `await` here to make the behavior clearer (i.e. that this function doesn't return a promise that resolves to a promise, it just returns an flattened promise).

Reply:
I think my silly self put the async await in the wrong place =/. ill remove it on line 113 and put an await here to make the logic a bit clearer